src/document/delta/transaction-base.js

```diff
@@ -0,0 +1,139 @@
+/**
+ * @license Copyright (c) 2003-2015, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+'use strict';
+
+CKEDITOR.define( [ 'ckeditorerror' ], ( CKEditorError ) => {
+	/**
+	 * The transaction class groups document changes (deltas). All deltas grouped in a single transactions can be
+	 * reverted together, so you can think about the transaction as a single undo step. If you want to extend one
+	 * undo step you can call another method on the same transaction object. If you want to create a separate undo step
+	 * you can create a new transaction.
+	 *
+	 * For example to create two separate undo steps you can call:
+	 *
+	 *	  doc.makeTransaction().insert( firstPosition, 'foo' );
+	 *	  doc.makeTransaction().insert( secondPosition, 'bar' );
+	 *
+	 * To create a single undo step:
+	 *
+	 *	  const transaction = doc.makeTransaction()
+	 *	  transaction.insert( firstPosition, 'foo' );
+	 *	  transaction.insert( secontPosition, 'bar' );
+	 *
+	 * Note that all document modification methods (insert, remove, split, etc.) are chainable so you can shorten code to:
+	 *
+	 *	  doc.makeTransaction().insert( firstPosition, 'foo' ).insert( secontPosition, 'bar' );
+	 *
+	 * @class document.Transaction
+	 */
+	class Transaction {
+		/**
+		 * Creates transaction instance. Not recommended to use directly, use {@link document.Document#makeTransaction}
+		 * instead.
+		 *
+		 * @constructor
+		 * @param {document.Document} doc Document which this transaction changes.
+		 */
+		constructor( doc ) {
+			/**
+			 * Document which this transaction changes.
+			 *
+			 * @readonly
+			 * @type {document.Document}
+			 */
+			this.doc = doc;
+
+			/**
+			 * Array of deltas which compose transaction.
+			 *
+			 * @readonly
+			 * @type {document.delta.Delta[]}
+			 */
+			this.deltas = [];
+		}
+
+		/**
+		 * Adds delta to the transaction instance. All modification methods (insert, remove, split, etc.) use this method
+		 * to add created deltas.
+		 *
+		 * @param {document.delta.Delta} delta Delta to add.
+		 *
+		 * @return {document.delta.Delta} Added delta.
+		 */
+		addDelta( delta ) {
+			delta.transaction = this;
+			this.deltas.push( delta );
+
+			return delta;
+		}
+
+		/**
+		 * Transaction provides iterator interface which will iterate over deltas in the transaction.
+		 */
+		[ Symbol.iterator ]() {
```

Comment:
So if we consider that the internals may change, then `this.delta` should be private. In other words – you need the iterator only if you cannot access the array.